src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -35,6 +36,8 @@ public LinterAnalyzer()
 
         private bool LinterVerbose(SemanticModel model) => model.Configuration.Analyzers.GetValue(LinterVerboseSetting, false);
 
+
+        [UnconditionalSuppressMessage("Trimming", "IL2072:Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The return value of the source method does not have matching annotations.", Justification = "List of types comes from a source generator.")]
```

Comment:
Is there any way to better inform the compiler about what's going on (without having to add a suppression)?

Reply:
The other trimming annotations rely on the type being discoverable via static analysis. From the analyzer's point of view, the list of types here is calculated at runtime (even if the hard-coded list is generated via a source generator). For this one I'm going to remove the reflection entirely and have the source generator generate constructor calls instead. This will be in a separate PR but will remove the need for suppression.